client/pages/projects/[projectId]/index.jsx

```diff
@@ -0,0 +1,160 @@
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import { Gantt, Head, TipCard } from "../../../components";
+import {
+  Button,
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalHeader,
+  Nav,
+  NavItem,
+  NavLink,
+  TabContent,
+  TabPane
+} from "reactstrap";
+import classnames from "classnames";
+import mockData from "../../../utils/mockData";
+
+import "../../../public/styles/project.scss";
+
+const DESCRIPTION_LENGTH = 400;
+
+const capitalize = str =>
+  str.length > 0 ? str.charAt(0).toUpperCase() + str.slice(1) : str;
+
+export default function ProjectPage() {
+  const [width, setWidth] = useState(null);
+  const [activePhase, setActivePhase] = useState("inspiration");
+  const [activeStage, setActiveStage] = useState(null);
+  const [modal, setModal] = useState(false);
+  const [project, setProject] = useState(null);
+  const [ganttData, setGanttData] = useState(null);
+
+  const router = useRouter();
+  const { projectId } = router.query;
+  const { projects } = mockData;
+
+  const toggleModal = () => setModal(!modal);
+
+  if (process.browser) {
+    useEffect(() => setWidth(document.body.clientWidth), [
+      document.body.clientWidth
+    ]);
+  }
+
+  useEffect(() => {
+    if (projectId < projects.length) {
+      setProject(projects[projectId]);
+    }
+  }, [projectId]);
+
+  const mapGanttData = phase =>
+    project.sections[phase.toLowerCase()]?.stages.map(stage => [
+      `${stage.name}-${phase}-${stage.description}`,
+      stage.name,
+      capitalize(phase),
+      new Date(stage.startdate),
+      new Date(stage.enddate),
+      null,
+      Math.random() * 100,
```

Comment:
What's the purpose of this random value?

Reply:
this is the progress (percentage done) - i'm not sure how that will be integrated into the workflow (if at all), so I just assigned it a random value as proof of concept